### PR TITLE
Add option to run the app under a non-root URL.

### PR DESCRIPTION
**Pull request by samuelsadok**
This PR addresses #168.

The `general.url_prefix` configuration option is added to run filestash at a location that is not the root of the domain. The prefix should be specified with a leading slash and no trailing slash. Leaving this option empty corresponds to previous behavior.

The option `general.url_prefix` can also be set though the `URL_PREFIX` environment variable.

The backend rewrites `index.html` to include a `<base>` tag which informs the frontend about the url prefix. The downside of this approach is that `index.html` cannot be cached or compressed.

This feature should be considered experimental. I have tested the basic functionality but I can't guarantee that everything will still work so it probably requires more testing.

### CODE REVIEW

**Comment by mickael-kerjean** [comment]
Hey, thank you for the PR! I don't have much time this week an the PR is quite thick, will take some time to properly review

**Comment by mickael-kerjean** [comment]
had a quick look, and it looks awesome though!

**Comment by samuelsadok** [comment]
Yeah no hurry!

**Comment by timaa2k** [comment]
I also wanted to file an issue for this. I'm trying to run `filestash` on Kubernetes behind a Traefik reverse proxy under `<domain>/cloud`.
Anything I can help with to get this merged?

**Review by devdreamsolution — APPROVED** [review]

**Comment by rlopes-sky** [comment]
Hello, it would be really nice if this option could be merged to master, we are really needing this. Do you know if this will be merged any time soon?

File: client/manifest.json [hunk 1]
```diff
@@ -3,12 +3,12 @@
     "short_name": "Filestash",
     "icons": [
         {
-            "src": "/assets/logo/android-chrome-192x192.png",
+            "src": "logo/android-chrome-192x192.png",
```
Comment (fdomingues-sky) [inline client/manifest.json:6]:
```suggestion
            "src": "assets/logo/android-chrome-192x192.png",
```

File: client/manifest.json [hunk 2]
```diff
@@ -3,12 +3,12 @@
...
             "type": "image/png",
             "sizes": "192x192"
         },
         {
-            "src": "/assets/logo/android-chrome-512x512.png",
+            "src": "logo/android-chrome-512x512.png",
```
Comment (fdomingues-sky) [inline client/manifest.json:11]:
```suggestion
            "src": "assets/logo/android-chrome-512x512.png",
```

**Comment by AnthonyRcd** [comment]
Any chance this gets merged one day ?

**Comment by NickSica** [comment]
Why was this pull request killed off by the stale bot? Was this functionality eventually added?

**Pull request closed without merging**